multinet/multinet/__init__.py

```diff
@@ -15,14 +15,32 @@
 from .schema import schema
 from . import db
 
+
+def graphql_query(query):
+    result = graphql(schema, query)
+    if result:
+        errors= [error.message for error in result.errors] if result.errors else []
+        logprint("Errors in request: %s" % len(errors), level=logging.WARNING)
+        for error in errors[:10]:
+            logprint(error, level=logging.WARNING)
+    else:
+        errors = []
+    return dict(data=result.data, errors=errors)
+
+
 class MultiNet(Resource):
     def __init__(self, port):
         super(MultiNet, self).__init__()
         self.resourceName = 'multinet'
         self.arango_port = port
         self.route('POST', ('graphql',), self.graphql)
         self.route('POST', ('bulk', ':workspace', ':table'), self.bulk)
-        self.route('POST', ('tree', ':workspace', ':table'), self.tree)
+
+        # Newick tree operations.
+        self.route('POST', ('newick', 'tree', ':workspace', ':table'), self.tree)
```

Comment:
A newick is a tree, so we don't need to re-specify that. I had used 'tree' when I thought we had a reader that handled multiple file formats that all essentially boiled down to being trees. I would advocate for keeping a form of this, and altering the bulk load endpoint to be a csv endpoint. We could also hypothetically use non-standard mime types, but that is discouraged.

So:
self.route('POST', ('newick', ':workspace', ':table'), self.tree)
self.route('POST', ('csv', ':workspace', ':table'), self.tree)

Reply:
Done in 1a8dae4.